examples/react-apollo/src/profile/security-keys.tsx

```diff
@@ -1,12 +1,64 @@
+import { useState } from 'react'
 import { FaMinus } from 'react-icons/fa'
+import { RemoveSecurityKeyMutation, SecurityKeysQuery } from 'src/generated'
 
+import { gql, useMutation } from '@apollo/client'
 import { ActionIcon, Button, Card, SimpleGrid, Table, TextInput, Title } from '@mantine/core'
 import { useInputState } from '@mantine/hooks'
-import { useSecurityKeys } from '@nhost/react'
+import { useAddSecurityKey, useUserId } from '@nhost/react'
+import { useAuthQuery } from '@nhost/react-apollo'
+
+const SECURITY_KEYS_LIST = gql`
+  query securityKeys($userId: uuid!) {
+    authUserAuthenticators(where: { userId: { _eq: $userId } }) {
+      id
+      nickname
+    }
+  }
+`
+
+const REMOVE_SECURITY_KEY = gql`
+  mutation removeSecurityKey($id: uuid!) {
+    deleteAuthUserAuthenticator(id: $id) {
+      id
+    }
+  }
+`
 
 export const SecurityKeys: React.FC = () => {
-  const { list, add, remove } = useSecurityKeys()
+  const { add } = useAddSecurityKey()
+  const userId = useUserId()
   const [nickname, setNickname] = useInputState('')
+  const [list, setList] = useState<{ id: string; nickname?: string | null }[]>([])
+  useAuthQuery<SecurityKeysQuery>(SECURITY_KEYS_LIST, {
+    variables: { userId },
+    onCompleted: ({ authUserAuthenticators }) => {
+      if (authUserAuthenticators) {
+        setList(authUserAuthenticators || [])
+      }
+    }
+  })
+
+  const addKey = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault()
+    const { key, error } = await add(nickname)
+    if (error) {
+      console.log(error)
+    } else {
+      setNickname('')
+    }
+    if (key) {
+      setList([...list, key])
+    }
```

Comment:
Isn't `key` always set if there is no error?

If so, this can be replaced with something like:


```js
  if (error) {
    return console.log(error)
  } 

  setNickname('')
  setList([...list, key])
```

Reply:
The reason was not just to support the docgen, but because it's generally a better practice to rely on interfaces as long as types are not needed. If type aliases make more sense, then go for type aliases.

This is what the official TS documentation also suggests: https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#differences-between-type-aliases-and-interfaces